### ISSUE TITLE
Are the weights optimized by the autograd of torch?

### ISSUE THREAD

**Issue by kevin031060** [ddehueck/pytorch-neat#4]
Hi, I understand that the topology is optimized through EA. How the weights are optimized. EA method or backpropagating by torch?

**Comment by ddehueck** [comment]
The weights are optimized in an evolutionary fashion as well. 

Note the following in config files:
```python
# Float between 0.0 and 1.0 - rate at which a connection gene will be mutated
CONNECTION_MUTATION_RATE = 0.80
# Float between 0.0 and 1.0 - rate at which a connections weight is perturbed (if connection is to be mutated) 
CONNECTION_PERTURBATION_RATE = 0.90
```

**Comment by ddehueck** [comment]
It should be possible to also update the weights with some training procedure before fitness evaluation. It could be interesting to see if something like the [Baldwin Effect](https://en.wikipedia.org/wiki/Baldwin_effect) is evident in the resulting fitness distribution.

**Comment by kevin031060** [comment]
Thank you and it's really a cool job. 
Is there any possibility that uses backpropagating method to train the generated network after using EA to determine its structure.  In other words, can we use the common backpropagating method to train an arbitrarily connected network? I think it can be more efficient if we use the backpropagating.

**Comment by ddehueck** [comment]
Yes, that would absolutely be possible. I'd be curious to see your results if you do so - we could add such an experiment to this repo perhaps.

**Comment by kevin031060** [comment]
I'll try it and hope there would be some progress.
